types/node.go

```diff
@@ -10,14 +10,17 @@ import (
 )
 
 type NodeConfig struct {
-	Name string
+	Name  string
+	Index int
```

Comment:
Something we should be doing is adding godoc comments to all exported types and fields.

Specifically for the `Index` field, which is a bit confusing what it is for.